src/GameSrc/audiolog.c

```diff
@@ -37,315 +34,235 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "faketime.h"
 #include "map.h"
 #include "tools.h"
-//#include "MacTune.h"
 #include "musicai.h"
 #include "mainloop.h"
-
 #include "bark.h"
 #include "miscqvar.h"
 
-#include <SDL_mixer.h>
+#include <SDL.h>
+
+
+
+#define AUDIOLOG_BASE_ID       2741
+#define AUDIOLOG_BARK_BASE_ID  3100
+
+#define MIN_ALOG_SIZE   20000
+
+#define ALOG_MUSIC_DUCK  0.7
+
+
+
+extern SDL_AudioStream *cutscene_audiostream; //see cutsloop.c
+
+static uint8_t *audiolog_audiobuffer = NULL;
+static uint8_t *audiolog_audiobuffer_pos = NULL;
+static int audiolog_audiobuffer_size; //in blocks of MOVIE_DEFAULT_BLOCKLEN
+
 
-//-------------------
-//  PROTOTYPES
-//-------------------
-uchar audiolog_cancel_func(short, ulong, void *);
 
-//-------------------
-//  GLOBALS
-//-------------------
 int curr_alog = -1;
-int alog_handle = -1;
 int alog_fn = -1;
 uchar audiolog_setting = 1;
-//Movie alog = NULL;
-
-Uint8 *alog_buffer = NULL;
-snd_digi_parms *sdp = NULL;
+char secret_pending_hack;
 
-// this can become RES_alog_email0 once there is one
-#define AUDIOLOG_BASE_ID 2741
-#define AUDIOLOG_BARK_BASE_ID 3100
+char *bark_files[] = { "res/data/citbark.res", "res/data/frnbark.res", "res/data/gerbark.res" };
+char *alog_files[] = { "res/data/citalog.res", "res/data/frnalog.res", "res/data/geralog.res" };
 
-#define MIN_ALOG_SIZE   20000
 
-//#define AUDIOLOG_BLOCK_LEN MOVIE_DEFAULT_BLOCKLEN
-//#define AUDIOLOG_BUFFER_SIZE  (64 * 1024)
-// uchar audiolog_buffer[AUDIOLOG_BUFFER_SIZE];
 
 extern uchar curr_vol_lev;
+extern uchar curr_alog_vol;
+extern char which_lang;
+
+
+
 extern void MacTuneUpdateVolume(void);
 
-#define ALOG_MUSIC_DUCK  0.7
 
-//-------------------------------------------------------------
-//  Sorry excuse for a function.
```

Comment:
This is a stylistic choice, but it's fun to see some of these original comments from the code. In the future, I'd err towards keeping things like this over throwing them away just for posterity's sake.